gaps/gap-5_payload_manifest/gap-5_payload_manifest.md

```diff
@@ -0,0 +1,127 @@
+---
+gap: 5
+title: Computation Payload Manifest
+description: Replacement for the `golem.srv.comp.task_package` property
+author: mf (@mfranciszkiewicz)
+status: Draft
+type: Feature
+requires (*optional): <GAP number TBD>
+---
+
+## Abstract
+
+The GAP proposes a new `golem.srv.comp.payload` property aiming to replace the `golem.srv.comp.task_package`. 
+The `payload` namespace covers URLs of computation payloads for multiple hardware architectures, an optional Computation
+Manifest ([GAP](https://github.com/golemfactory/golem-architecture/pull/27)) and also an optional Payload Manifest 
+signature.
+
+## Motivation
+
+The main motivation behind the proposed changes is to provide a source of trust for the executed Payload. The Manifest 
+may include a default, safe set of actions Requestors can execute on a Provider's machine, approved by a trusted 
+Authority. This source of trust allows for implementing high-risk features, such as Internet access for Runtimes, where 
+all traffic is routed via Provider's OS. In order to prevent unwanted actions in a public network on Provider's behalf, 
+she must remain in control of all the actions executed by the Requestor.
+
+The secondary objective is to provide multiple locations of Payloads, targeting different hardware architectures and 
+meeting specific software constraints.
+
+None of the above are possible with the `golem.srv.comp.task_package` property.
+
+## Specification
+
+Computation Payload Manifest signatures are verified by either the Provider Agent, the ExeUnit Supervisor or both.
+Payload and Computation manifests are not expected to have constraints put on them. 
+
+### `golem.srv.comp.payload: String`
+
+Base64-encoded JSON manifest.
```

Comment:
Do we need to make the `golem.srv.comp.payload` Base64-encoded? 
It is a JSON, so technically there shouldn't be a need for encoding. Leaving it unencodedwould make 'visual troubleshooting' easier...

Reply:
It doesn't need to be base64-encoded, a JSON string would be enough. It will still be difficult to read, though